### PR TITLE
Use "viewport" meta tag in docs

### PR DESCRIPTION
**Pull request by chaficnajjar**
As proposed by @eriwen in gradle/dotorg-docs#25, I used the ["viewport" meta tag](https://developer.mozilla.org/en-US/docs/Mozilla/Mobile/Viewport_meta_tag) in the `<head>` of the HTML pages for the release notes, user guide and DSL reference.

Two reasons for this change:

### 1. The release notes, user guide and DSL reference are now more readable on mobile devices

For example, here's a before-after of the release notes on an iPhone 6.

#### Before

screenshot

#### After

screenshot

### 2. These pages will become mobile-friendly on https://search.google.com/test/mobile-friendly

I can't test this now, but I think the following 4 issues will be resolved once this change makes it to https://docs.gradle.org.

screenshot

### CODE REVIEW

**Comment by eriwen** [comment]
Thank you @chaficnajjar. I've verified the changes and cherry picked these commits into release. They will be merged to master shortly.

File: subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl [hunk 1]
```diff
@@ -32,9 +32,9 @@
     <xsl:param name="css.decoration">0</xsl:param>
     <xsl:param name="highlight.source" select="1"/>
 
-    <!-- Use custom style sheet content -->
-    <xsl:param name="html.stylesheet">DUMMY</xsl:param>
-    <xsl:template name="output.html.stylesheets">
+    <!-- Customize the metadata and stylesheets to add to the <head> element -->
+    <xsl:template name="head.content">
```
Comment (eriwen) [inline subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl]: I'm afraid this completely overwrites the current head content, and a bunch of other headers were now omitted. I've reverted this commit.

Comment (chaficnajjar) [reply subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl]: > I'm afraid this completely overwrites the current head content, and a bunch of other headers were now omitted.

@eriwen You're right, thanks for pointing that out. In the user guide, my changes removed the following two lines:

```
<title>Gradle User Guide Version 4.4-20171127220000+0000</title>
<meta content="DocBook XSL Stylesheets V1.75.2" name="generator">
``` 

The same lines are also removed in the DSL reference. I'll try to see if I can keep these two lines along with the `<meta>` tag I added.

## User guide <head> after your revert (i.e. before my changes)

```
git checkout dc30798574d99957cec500a893f00f78890577a7
./gradlew docs:userguide
```

screenshot

## User guide <head> before your revert (i.e. after my changes)

```
git checkout dc30798574d99957cec500a893f00f78890577a7^
./gradlew docs:userguide
```

screenshot

Comment (chaficnajjar) [reply subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl]: @eriwen My bad, it seems like you've already fixed the issue: 0a77f4b4b67563c569bd55a518446637e8349cb8. However, this commit only adds the `viewport` meta tag to the user guide and not the DSL reference. But we discussed this privately and decided that we didn't want to add the `viewport` meta tag to the DSL reference, so this change is fine and no more changes are required.

**Pull request closed without merging**